lib/editor/components/pattern/EditShapePanel.js

```diff
@@ -188,6 +227,7 @@ export default class EditShapePanel extends Component<Props> {
     const nextSegment = (!patternSegment && patternSegment !== 0)
       ? 0
       : patternSegment + 1
+    const patternStopsWithShapeIssues = this._getPatternStopsWithShapeIssues()
```

Comment:
It's kind of impressive that this doesn't seem to be a major performance hit even for a pattern with 140 stops.